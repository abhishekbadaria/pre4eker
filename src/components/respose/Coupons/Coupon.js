import React from "react";
import PropTypes from "prop-types";
import stylesShared from "src/components/respose/responseShared.css";
import statuses from "./statuses";

const Coupon = ({ couponInfo }) => {
  const { discountAmountForCurrentOrder } = couponInfo;

  const {
    status,
    availableFromDateTimeUtc,
    availableTillDateTimeUtc,
    usedDateTimeUtc,
  } = couponInfo.coupon;

  const { code } = couponInfo.coupon.ids;

  const availableFrom = new Date(availableFromDateTimeUtc);
  const availableTill = new Date(availableTillDateTimeUtc);
  const usedDate = new Date(usedDateTimeUtc);

  return (
    <div className={stylesShared.promo}>
      <div className={stylesShared.promoType}>Промокод</div>
      <div className={stylesShared.inline}>
        <div className={stylesShared.half}>
          <div className={stylesShared.line}>
            <div className={stylesShared.value}>{code}</div>
          </div>
          <div className={stylesShared.line}>{statuses[status]}</div>
          <div className={stylesShared.line}>
            <div className={stylesShared.name}>Сумма скидки</div>
            <div className={stylesShared.promoValue}>
              {discountAmountForCurrentOrder}
            </div>
          </div>
        </div>
        <div className={stylesShared.half}>
          {availableFromDateTimeUtc && (
            <div className={stylesShared.line}>
              <div className={` ${stylesShared.name}`}>Выдан</div>
              <div className={`  ${stylesShared.value}`}>
                {availableFrom.toDateString()}
              </div>
            </div>
          )}
          {availableTillDateTimeUtc && (
            <div className={stylesShared.line}>
              <div className={`  ${stylesShared.name}`}>Сгорит</div>
              <div className={`  ${stylesShared.value}`}>
                {availableTill.toDateString()}
              </div>
            </div>
          )}
          {usedDateTimeUtc && (
            <div className={stylesShared.line}>
              <div className={` ${stylesShared.name}`}>Использован</div>
              <div className={`  ${stylesShared.value}`}>
                {usedDate.toDateString()}
              </div>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

Coupon.propTypes = {
  couponInfo: PropTypes.object,
};
Coupon.defaultProps = {
  couponInfo: {},
};

export default Coupon;
