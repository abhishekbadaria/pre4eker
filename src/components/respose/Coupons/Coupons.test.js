import React from "react";
import { render, cleanup } from "@testing-library/react";
import Coupon from "./Coupon";

afterEach(cleanup);

const coupon = {
  discountAmountForCurrentOrder: 10528369.9,
  coupon: {
    ids: {
      code: "ABC12345",
    },
    pool: {
      ids: {
        mindboxId: 683788363,
        externalId: "pool-1",
      },
      name: "Новый пул промокодов",
      description: "Пул для новой акции!",
    },
    status: "YetNotActive",
    availableFromDateTimeUtc: "2018-10-22T22:04:40.76Z",
    availableTillDateTimeUtc: "2017-11-20T22:04:40.76Z",
    usedDateTimeUtc: "2019-04-19T22:04:40.76Z",
  },
};

test("customer match snapshot", () => {
  const { asFragment } = render(<Coupon couponInfo={coupon} />);
  expect(asFragment(<Coupon />)).toMatchSnapshot();
});
