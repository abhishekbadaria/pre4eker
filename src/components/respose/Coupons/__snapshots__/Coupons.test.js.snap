// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer match snapshot 1`] = `
<DocumentFragment>
  <div
    class="promo"
  >
    <div
      class="promoType"
    >
      Промокод
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="line"
        >
          <div
            class="value"
          >
            ABC12345
          </div>
        </div>
        <div
          class="line"
        >
          Пока еще не активен
        </div>
        <div
          class="line"
        >
          <div
            class="name"
          >
            Сумма скидки
          </div>
          <div
            class="promoValue"
          >
            10528369.9
          </div>
        </div>
      </div>
      <div
        class="half"
      >
        <div
          class="line"
        >
          <div
            class=" name"
          >
            Выдан
          </div>
          <div
            class="  value"
          >
            Tue Oct 23 2018
          </div>
        </div>
        <div
          class="line"
        >
          <div
            class="  name"
          >
            Сгорит
          </div>
          <div
            class="  value"
          >
            Tue Nov 21 2017
          </div>
        </div>
        <div
          class="line"
        >
          <div
            class=" name"
          >
            Использован
          </div>
          <div
            class="  value"
          >
            Sat Apr 20 2019
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
